spack-repo/packages/spiner/package.py

```diff
@@ -64,8 +63,8 @@ class Spiner(CMakePackage):
     depends_on("ports-of-call portability_strategy=Kokkos", when="@:1.5.1 +kokkos")
     depends_on("ports-of-call portability_strategy=None", when="@:1.5.1 ~kokkos")
     depends_on("kokkos@3.3.00:", when="+kokkos")
-    depends_on(
-        "kokkos ~shared+cuda_lambda+cuda_constexpr",
+    requires(
+        "^kokkos ~shared+cuda_lambda+cuda_constexpr",
```

Comment:
Interesting... I saw in  your other MR something about an "audit failure". Could you expand on this change? Or perhaps more specifically, can you let us know when we should use `requires` versus `depends_on`?

Reply:
See this conversation in the Spack Slack: https://spackpm.slack.com/archives/C5W7NKZJT/p1736452754080359

Reply:
In summary: there is currently an audit rule for Spack packages that are contributed to Spack itself. It fails when there are `depends_on` lines where the requirements is referencing itself in the `when=` clause. The audit rule is overly aggressive, so you can work around it via the `requires` directive.